### PR TITLE
[Feature] Support external catlog -- add getUUID api

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Change the id of DbPEntryObject and TablePEntryObject to uuid in
string format because for external catalog, the connector cannot
generate a unique integer id for its table or database. For internal
table or db, we just use the interger id directly, SR already ensure its
uniqueness, for external table or, it's up to implementation of connector.

Signed-off-by: Dejun Xia <xiadejun@starrocks.com>

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/privilege/DbPEntryObject.java [hunk 1]
```diff
@@ -23,13 +23,17 @@
 import java.util.Objects;
 
 public class DbPEntryObject implements PEntryObject {
-    public static final long ALL_DATABASE_ID = -2; // -2 represent all
+    public static final String ALL_DATABASES_UUID = "ALL_DATABASES_UUID"; // -2 represent all
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/privilege/DbPEntryObject.java:26]: why this change to string? what comment "-2 represent all" means?

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by nshangyiming** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/privilege/DbPEntryObject.java (hunk 1 above)
Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/privilege/DbPEntryObject.java:26]: In order to be in compatible with external table, see the comments for this pr. the comments need to be modified, i will change it.

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 70 / 83 (84.34%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|4|9|44.44%|[342, 343, 364, 368, 370]|
|:large_blue_circle:|com/starrocks/privilege/DbPEntryObject.java|15|22|68.18%|[32, 58, 59, 61, 108, 112, 132]|
|:large_blue_circle:|com/starrocks/privilege/TablePEntryObject.java|33|34|97.06%|[159]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/MaterializedViewPEntryObject.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/ViewPEntryObject.java|7|7|100.00%|[]|

**Comment by nshangyiming** [comment]
sh starrocks_admit_test

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test
